packages/task/src/browser/quick-open-task.ts

```diff
@@ -182,6 +182,7 @@ export class QuickOpenTask implements QuickAccessProvider {
         picker.matchOnDescription = true;
         picker.ignoreFocusOut = false;
         picker.items = this.items;
+        picker.onDidTriggerItemButton(({ item }) => this.onDidTriggerGearIcon(item as QuickPickItem));
```

Comment:
It seems odd that we have to case here, but not below. Any idea why?

Reply:
The two methods have different signatures :smile: 
Here the item can be a separator as well, even if it won't happen (separators  won't have buttons that will trigger this method). However, to make this nicer I updated the code and added a check here, instead.

Reply:
That was the question: is it true that one time the button item can be a separator as well or is that a mistake in the typing of "picker.onDidTriggerItemButton()"?

Reply:
Hi @tsmaeder! No, the item passed to the button cannot be a separator. The separator only has a label and no buttons. I restricted the type in the interface for this method. Ideally, we would have a dedicated type for a separator picker, that does not have this method at all, but I think that refactoring is out of scope for this PR.